static/src/javascripts/projects/common/utils/steady-page.js

```diff
@@ -0,0 +1,187 @@
+                    });
+                });
+            })
+            .then(function(state){
+                if (q.empty()) {
+                    // If the queue is empty (no more elements need to be added to the page) we immediately scroll
+                    return state.newHeight + state.prevHeight + state.scrollY;
+                } else {
+                    // If there are elements waiting to be added to the page we take the previous container's heights
+                    // and recursively call the function so that we only scroll the page once the queue is empty -
+                    // this prevents excessive and jarring scrolling
+                    return go(assign(state, {
+                        prevHeight: state.prevHeight + state.newHeight
+                    }));
+                }
+            });
+
+        return promise;
+
+    }
+
+    /**
+     * Insert an element into the page
+     * Use if your element doesn't exist and is inserted into a container
+     * ** Don't use fastdom - it is handled in this utility **
+     * @param {HTMLElement} container The element that the component is being inserted into
+     * @param {Function} cb Should contain all functionality that displays and lays-out the element
+     */
+    function insert(container, cb) {
+        if (!config.switches.steadyPageUtil) {
+            return fastdom.write(cb);
+        }
+
+        var initialState = {
+            scrollY: window.scrollY,
+            prevHeight: 0
+        };
+
+        q.enqueue({
+            container: container,
+            cb: cb
+        });
+
+        return (running ? promise : go(initialState)).then(scrollThePage);
+    }
+
+
+    /**
+     * Given a batch, call all of the callbacks on the insertion object
+     *
+     * @param  {Array} batch
+     */
+    function insertElements (batch) {
+        return fastdom.write(function () {
+            batch.forEach(function (insertion) {
+                insertion.cb();
+            });
+        });
+    }
+
+    function scrollThePage (scrollY) {
+        if (scrollY) {
+            window.scrollTo(0, scrollY);
+            running = false;
+        }
+    }
+
+    /**
+     * Given a batch and a previous currentBatchHeight, measure the height of each container
+     * in the batch
+     *
+     * @param  {Array} batch
+     */
+    function getHeightOfAllContainers (batch) {
+        return fastdom.read(function() {
+            // Add all the heights of the passed in batch
+            // removing the current height
+            return batch.filter(elementIsAbove).reduce(function(height, insertion) {
+                return height + readHeight(insertion.container);
+            }, 0);
+        });
+
+        function elementIsAbove(el) {
+            var parentElArray = [];
+            var parentEl = el.container;
+            var viewportAdjustment = Math.max(document.documentElement.clientHeight, window.innerHeight || 0) / 2;
```

Comment:
You can move this line into the `fastdom.read` callback so it is computed only once.